### PR TITLE
[CID-1842] - Allow user-managers to view/edit other user-managers

### PR DESCRIPTION
**Pull request by kidster**

### CODE REVIEW

**Comment by rackernexus** [comment]
Can one of the admins verify this patch?

File: api-docs/api-reference/methods/roles/delete-global-role-from-user-v2.0.rst [hunk 1]
```diff
@@ -19,6 +19,19 @@ revoking  access at will. Some roles like identity:rack_connect and
...
 revoking privileges unintentionally.
 
+.. note::
+
+  - Users with the ``identity:service-admin`` role can delete a global role
+    from a user for users with the ``identity:admin`` role, users with the
+    ``identity:user-admin`` role, and sub-users.
+
+  - Users with the ``identity:admin`` role can delete a global role from a user
+    for users with the ``identity:user-admin`` role, and sub-users.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/roles/delete-global-role-from-user-v2.0.rst]: role and sub-users. (no comma needed)

File: api-docs/api-reference/methods/roles/put-add-role-to-user-v2.0.rst [hunk 2]
```diff
@@ -19,6 +19,19 @@ If an Identity service Administrator (``identity:service-admin``) assigns a role
...
 users (``identity:default``).
 
+.. note::
+
+  - Users with the ``identity:service-admin`` role can add a role to a user for
+    users with the ``identity:admin`` role, users with the
+    ``identity:user-admin`` role, and sub-users.
+
+  - Users with the ``identity:admin`` role can add a role to a user for users
+    with the ``identity:user-admin`` role, and sub-users.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/roles/put-add-role-to-user-v2.0.rst]: role and (no comma needed)

File: api-docs/api-reference/methods/roles/put-grant-roles-to-user-v2.0.rst [hunk 3]
```diff
@@ -12,17 +12,33 @@ being modified remain unchanged.
...
-   - The caller must be authorized to assign the specified roles.
-   - The tenant assignments can include either ["*"] or a set of tenant ids.
-   - This request effectively replaces any assignment that already exists for
-     these roles on the user.
+  - Users with the ``identity:service-admin`` role can grant roles to a user
+    for users with the ``identity:admin`` role, users with the
+    ``identity:user-admin`` role, and sub-users.
+
+  - Users with the ``identity:admin`` role can grant roles to a user for users
+    with the ``identity:user-admin`` role, and sub-users.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/roles/put-grant-roles-to-user-v2.0.rst]: role and (no comma needed)

File: api-docs/api-reference/methods/users/delete-a-user-v2.0.rst [hunk 4]
```diff
@@ -16,6 +16,22 @@ To delete a user, specify the user ID in the request. If you know the user name
...
+
+  - Users can always use this service to delete themselves with the exception
+    of ``identity:user-admin`` users.
+
+  - Users with the ``identity:service-admin`` role can delete users with the
+    ``identity:admin`` role, users with the ``identity:user-admin`` role, and
+    sub-users.
+
+  - Users with the ``identity:admin`` role can delete users with the
+    ``identity:user-admin`` role, and sub-users.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/delete-a-user-v2.0.rst]: role and (no comma needed)

File: api-docs/api-reference/methods/users/post-reset-api-key-for-user-v2.0-users-userid-os-ksadm-credentials-rax-kskey.rst [hunk 5]
```diff
@@ -18,6 +18,21 @@ updating the password,  the API key reset does not revoke existing tokens.
...
+
+  - Users can always use this service to reset the API key credentials for
+    themselves.
+
+  - Users with the ``identity:service-admin`` role can reset API key
+    credentials for users with the ``identity:admin`` role, users with the
+    ``identity:user-admin`` role, and sub-users.
+
+  - Users with the ``identity:admin`` role can reset API key credentials for
+    users with the ``identity:user-admin`` role, and sub-users.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-reset-api-key-for-user-v2.0-users-userid-os-ksadm-credentials-rax-kskey.rst]: role and (no comma needed)

**Pull request merged**